src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -820,6 +820,7 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 }
                 else // addrNode is used
                 {
+                    assert(!addrNode->isContained());
```

Comment:
We are asserting that `addrNode` is not contained, and yet calling `genConsumeAddress` which handles a contained `LEA`. Is that so that it will handle a contained address even if we don't expect it? If so, a comment would be good. If not, it should be calling `genConsumeReg()`.

Reply:
Added a comment about that.